src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -439,5 +440,94 @@ public static string MapNameToOID(string name)
 
             return oidName;
         }
+
+        static public byte[] EncodeOID(string str)
+        {
+            if (str == null)
+                throw new ArgumentNullException(nameof(str));
+
+            char[] sepArray = { '.' }; // valid ASN.1 separators
+            string[] oidString = str.Split(sepArray);
+            uint[] oidNums = new uint[oidString.Length];
+            for (int i = 0; i < oidString.Length; i++)
+            {
+                oidNums[i] = (uint)Int32.Parse(oidString[i], CultureInfo.InvariantCulture);
+            }
+
+            // Allocate the array to receive encoded oidNums
+            byte[] encodedOidNums = new byte[oidNums.Length * 5]; // this is guaranteed to be longer than necessary
+            int encodedOidNumsIndex = 0;
+            
+            // Handle the first two oidNums special
+            if (oidNums.Length < 2)
+                throw new CryptographicUnexpectedOperationException(SR.Cryptography_InvalidOID);
+
+            uint firstTwoOidNums = (oidNums[0] * 40) + oidNums[1];
+            byte[] retval = EncodeSingleOIDNum(firstTwoOidNums);
```

Comment:
Instead of `EncodeSingleOIDNum` returning a new `byte[]` each call and then copying that array into the destination, why not just have `EncodeSingleOIDNum` write directly to the destination array at the specified index?

Something like:

```c#
EncodeSingleOIDNum(firstTwoOidNums, retval, ref encodedOidNumsIndex);

private static void EncodeSingleOIDNum(uint value, byte[] destination, ref int index)
{
    // write directly to destination starting at index, then update index based on how many bytes written
}
```